jib-core/src/main/java/com/google/cloud/tools/jib/frontend/JavaDockerContextGenerator.java

```diff
@@ -339,11 +338,15 @@ String makeDockerfile() throws JsonProcessingException {
 
     dockerfile.append(mapToDockerfileString(environment, "ENV"));
     dockerfile.append(mapToDockerfileString(labels, "LABEL"));
-    dockerfile
-        .append("\nENTRYPOINT ")
-        .append(objectMapper.writeValueAsString(entrypoint))
-        .append("\nCMD ")
-        .append(objectMapper.writeValueAsString(programArguments));
+    if (entrypoint != null) {
+      dockerfile.append("\nENTRYPOINT ").append(objectMapper.writeValueAsString(entrypoint));
+    }
+    if (programArguments != null) {
+      dockerfile.append("\nCMD ").append(objectMapper.writeValueAsString(programArguments));
+    } else if (entrypoint != null) {
+      // Do not inherit CMD from base image if ENTRYPOINT is not inherited
+      dockerfile.append("\nCMD []");
```

Comment:
I think we don't need this. if `entrypoint` is not null, `Dockefile` will have the line `ENTRYPOINT` but no `CMD`, and Docker build won't inherit CMD.

Reply:
Ok, I've removed this.